### PR TITLE
Custom TransactionSigner and eth-account module improvements.

### LINKED ISSUE #2356: Call to account.signTransaction({...}) throw TypeError
#### Expected behavior
When supplying a fully defined transaction to `account.signTransaction`, a signed transaction should be returned.

#### Actual behavior
As it stands, the method throws a TypeError: this.signTransaction(...).bind is not a function. This is caused by the early return on line 230 of Accounts.js.

#### Steps to reproduce the behavior

```js
const Web3 = require("web3");

const web3 = new Web3(Web3.givenProvider || "ws://localhost:8546");
let pk = "0x0e652850717658cfd360d2c97474f4b2d8db5a9222adef2ebce67788f75e58f8";
let account = web3.eth.accounts.privateKeyToAccount(pk);
let signedTx = account.signTransaction({
    nonce: "0",
    chainId: "1",
    to: account.address,
    data: web3.utils.toHex('msg'),
    value: "0",
    gasPrice: "0",
    gas: "0"}
    );
```

#### Error Logs

`TypeError: this.signTransaction(...).bind is not a function`

#### Versions

* NPM: 6.4.1
* Node: 10.15.0
* Web3.js: 1.0.0-beta.44
* OS: Windows 10

### PR DESCRIPTION
**Pull request by nivida**
## Description

This PR fixes #2356 #2378 #1998 #1255 #2300 and will also improve the QA of the module.

### Web3 Accounts Refactoring

The eth-accounts module is one of the last modules which didn't got improved during the last refactoring of the core modules. 
My idea now is to clean up the eth-accounts module and to add a ``transactionSigner`` options property for the eth module. 

This sounds like a big refactoring but I will mostly just move the things. 

#### TransactionSigner
It will be possible to create your own class of type ``TransactionSigner`` and inject it over the options property. The class name ``TransactionSigner`` is a reserved class name and will be used for the internal signer class.  

```javascript

interface TransactionSigner {
  sign(txObject: Transaction): Promise<SignedTransaction>
}

interface SignedTransaction {
  messageHash: string,
  v: string,
  r: string,
  s: string,
  rawTransaction: string
}
```

##### Example
```javascript
const eth = new Eth(provider, {transactionSigner: new HDWalletSigner(...)});
```

#### Accounts 
The interface of the Accounts class (``web3.eth.accounts``) will still be the same. 

## Type of change

- [x] Bug fix 
- [x] Enhancement

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no warnings.
- [x] I have updated or added types for all modules I've changed
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran ```npm run test``` in the root folder with success and extended the tests if necessary.
- [x] I ran ```npm run build``` in the root folder and tested it in the browser and with node.
- [x] I ran ```npm run dtslint``` in the root folder and tested that all my types are correct
- [x] I have tested my code on the live network.

### CODE REVIEW

File: packages/web3-eth-contract/src/AbstractContract.js [hunk 1]
```diff
@@ -59,8 +61,10 @@ export default class AbstractContract extends AbstractWeb3Module {
         this.abiMapper = this.contractModuleFactory.createAbiMapper();
         this.options = options;
         this.PromiEvent = PromiEvent;
+        this.accounts = accounts;
         this.methodFactory = this.contractModuleFactory.createMethodFactory();
         this.abiModel = this.abiMapper.map(abi);
+        this.transactionSigner = options.transactionSigner;
         this.options = options;
```
Comment (sokun-chorn-webjet) [inline packages/web3-eth-contract/src/AbstractContract.js]: duplicates with line 62

Comment (nivida) [reply packages/web3-eth-contract/src/AbstractContract.js]: This isn't a duplication. I'm just passing the `TransactionSigner` dependency over the options. Check out the Eth module for seeing the real reason why I'm doing this. The complete dependency handling will get improved later (removing of the Factory objects etc.).

Comment (csokun) [reply packages/web3-eth-contract/src/AbstractContract.js]: I mean this assignment `this.options = options;`  it happens twice probably a left over from git merge.

Comment (nivida) [reply packages/web3-eth-contract/src/AbstractContract.js]: Oh sorry, I got it wrong. Thanks.

**Comment by rstormsf** [comment]
@nivida Any great example on how to use it?

File: packages/web3-core-method/src/watchers/NewHeadsWatcher.js [hunk 2]
```diff
@@ -47,10 +46,11 @@ export default class NewHeadsWatcher extends EventEmitter {
      * @returns {NewHeadsWatcher}
      */
     watch(moduleInstance) {
-        if (this.isSocketConnection(moduleInstance.currentProvider)) {
+        if (moduleInstance.currentProvider.constructor.name !== 'HttpProvider') {
             this.confirmationSubscription = this.subscriptionsFactory
                 .createNewHeadsSubscription(moduleInstance)
                 .subscribe(() => {
+                    console.log('newHEAD');
```
Comment (sebayaki) [inline packages/web3-core-method/src/watchers/NewHeadsWatcher.js:53]: maybe forgot to remove?

**Pull request merged**